packages/react-testing/src/tests/e2e.test.tsx

```diff
@@ -1,218 +1,471 @@
-import React from 'react';
+import React, {
+  useState,
+  useEffect,
+  useContext,
+  createContext,
+  ReactNode,
+  Component,
+  ComponentType,
+  memo,
+  forwardRef,
+  PureComponent,
+  Ref,
+} from 'react';
 import {createPortal} from 'react-dom';
 
-import {mount} from '../mount';
+import {mount, createMount} from '../mount';
 
-describe('e2e', () => {
-  it('can test the result of triggering props', () => {
-    function Clickable({onClick}: {onClick(): void}) {
-      return (
-        <button type="button" onClick={onClick}>
-          Click me
-        </button>
-      );
+describe('@shopify/react-testing', () => {
+  it('does not time out with large trees', () => {
+    function RecurseMyself({times}: {times: number}) {
+      if (times <= 0) {
+        return <div>finished</div>;
+      }
+      return <RecurseMyself times={times - 1} />;
     }
+    expect(() => {
+      mount(<RecurseMyself times={900} />);
+    }).not.toThrow();
+  });
 
-    function MyComponent() {
-      const [clicked, setClicked] = React.useState(false);
-
-      return clicked ? (
-        <div>Clicked!</div>
-      ) : (
-        <Clickable onClick={setClicked.bind(null, true)} />
-      );
-    }
+  it('can output structured debug strings', () => {
+    const wrapper = mount(
+      // eslint-disable-next-line @shopify/jsx-prefer-fragment-wrappers
```

Comment:
In web I disabled this for all test files. Would it be valuable to do that here?